app/jobs/pull_email_job.rb

```diff
@@ -0,0 +1,86 @@
+require 'logger'
+
+class PullEmailJob < ApplicationJob
+  queue_as :default
```

Comment:
This job probably needs a `before_perform` to schedule the job occurring on the next 20 minute interval. Or, see other comment below we can have a separate job scheduling this job that incorporates the IMAP idle notifications.